### PR TITLE
feat: create github k8s service account

### LINKED ISSUE #65: [CLOUD] Identify a method to generate credentials that can github action to deploy to the cluster.
### Issue Type

Enhancement

### Description

**The Context:**
At the current state, it is not easy at all to deploy this project on a remote cluster. We claim it is as easy as deploying Argo and applying our manifest but in fact, there are a series of challenges to overcome while doing that in order to achieve a perfectly working cluster, as we intended:
- In order to access the Kubernetes master API to provide a deployment context to the [Github Action](https://github.com/nearform-actions/k8s-kurated-addons-action) it is required to set up an OIDC connection, [for example for AWS](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services), it's not that easy to figure out which is the role to use, as well as which IAM permissions are required to make it work in the first place.

These are at the current state the challenges I faced with @LucaLanziani while we were trying to deploy this solution on an AWS EKS cluster and probably we missed some others that might happen down the road, but at the current state those are already some huge roadblocks we noticed that require immediate attention if we want to deliver an easy to use experience for DevOps onboarding this project, no matter their background.

**The Solution:**
First of all, it would be nice to identify if there is a way to abstract the OIDC connection layer to ensure it works cross-cloud and ideally even on bare metal. Once that has been figured out, an ideal solution would be to provide an easy-to-use script and/or a make command that would abstract this complexity and allow everyone to deploy this project on their target cloud environment ( AWS, Google, Azure, Alibaba, etc. ) with ease.

**The Alternatives:**
An alternative that could be considered but IMHO should be done meanwhile the full scripted solution is provided, is to create a series of documentation that would cover this aspect and help the DevOps operator to successfully deploy this solution on their intended cloud platform.

### Requirements

- These credentials will be used in GitHub action on developers repos.
- These credentials should only be able to deploy things on the cluster.

### Desired Outcome

- Have credential or tokens or a way to authenticate that could be used by a CLI tool running on Github actions

### Detailed steps

```bash
N/A
```

### Screenshots

```bash
N/A
```

### Logs

_No response_

### PR DESCRIPTION
**Pull request by blucas**
## What does this PR do?

Creates a service account that can be used by CI systems for authentication/authorization to the k8s cluster.  Also includes a temporary workflow which demonstrates that the service account can authenticate to the cluster and is authorized to execute various operations on said cluster.  This workflow does not need to be committed.

## Related issues

Closes #65 

## Checklist before merging

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have checked the [contributing document](../CONTRIBUTING.MD).
- [x] I have checked the existing [Pull Requests](https://github.com/nearform/k8s-kurated-addons/pulls) to see whether someone else has raised a similar idea or question.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have upgraded the [changelog](../CHANGELOG.md) according to the nature of the feature that I am adding to this Pull Request.

### CODE REVIEW

File: .github/workflows/k8s-access.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          aws eks update-kubeconfig --name k8s-kurated-addons
+          kubectl config set-credentials ci-bot --token=$K8S_GHA_TOKEN
+          kubectl config set-context k8s-kurated-addons-gha --cluster=arn:aws:eks:eu-west-1:$AWS_ACCOUNT_ID:cluster/k8s-kurated-addons --user=ci-bot
+          kubectl config use-context k8s-kurated-addons-gha
+          kubectl get namespaces
+          kubectl get pods
+
+        env:
+          K8S_GHA_TOKEN: ${{ secrets.DEVOPS_BOT_K8S_SERVICE_ACCOUNT_TOKEN }}
+          AWS_ACCOUNT_ID: ${{ secrets.DEVOPS_BOT_K8S_AWS_ACCOUNT_ID }}
```
Comment (LucaLanziani) [inline .github/workflows/k8s-access.yaml:41]: @blucas I don't think we will need this action in the repo, did you create this for testing the functionality?

File: scripts/create-ci-service-account.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+kubectl apply -f manifests/gha-service-account
```
Comment (LucaLanziani) [inline scripts/create-ci-service-account.sh:3]: I would probably add this as a target in the makefile

File: .github/workflows/k8s-access.yaml (hunk 1 above)
Comment (blucas) [reply .github/workflows/k8s-access.yaml:41]: correct, just to prove out that it works.  Once all other feedback is addressed, I'll remove the workflow prior to merging.

**Comment by LucaLanziani** [comment]
@blucas one for you ;)
```
➜  k8s-kurated-addons git:(feat/create-gh-k8s-service-account) make create-ci-service-account 
/bin/bash: line 1: ./scripts/create-ci-service-account.sh: Permission denied
make: *** [Makefile:96: create-ci-service-account] Error 126
```

**Comment by LucaLanziani** [comment]
@blucas another one please:
```
➜  k8s-kurated-addons git:(feat/create-gh-k8s-service-account) ✗ make create-ci-service-account
error: the path "manifests/gha-service-account" does not exist
make: *** [Makefile:96: create-ci-service-account] Error 1
```

File: manifests/ci-service-account/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Service account for CI Bot
```
Comment (zimny) [inline manifests/ci-service-account/README.md:1]: Can you add appropriate section to the main readme file, that will describe the use case for this manifest and how/when to use it + any dependencies?

Comment (blucas) [reply manifests/ci-service-account/README.md:1]: I can add some extra paragraphs about why this exists to the root README for sure.  One question though.
Do you want me to remove this readme from _this_ directory and place its contents in the README in the root repo?

**Review by zimny — APPROVED** [review]

**Pull request merged**